python/arcticdb/version_store/library.py

```diff
@@ -105,7 +106,7 @@ class NameWithDType(NamedTuple):
     """A named tuple. A name and dtype description pair."""
 
     name: str
-    dtype: str
+    dtype: TypeDescriptor
```

Comment:
This is great. Thank you for spotting this, and it looks correct to me. Please see the below though.

I noticed when testing your change that the `index` property of `SymbolDescription` actually returns a `NameWithDType` where the `dtype` is `List[TypeDescriptor]`. See below.

```python
>>> l
Library(Arctic(config=MEM()), path=l, storage=in_memory_storage)
>>> l.list_symbols()
['s']
>>> isinstance(l.get_description('s').index.dtype[0], arcticc.pb2.descriptors_pb2.TypeDescriptor)
True
>>> l.get_description('s').index.dtype
[value_type: INT
]
>>> isinstance(l.get_description('s').index.dtype[0], arcticc.pb2.descriptors_pb2.TypeDescriptor)
True
>>> isinstance(l.get_description('s').index.dtype, list)
True
```

This is due to in [`library.py`](https://github.com/man-group/ArcticDB/blob/904b0e5b88bdf9b7c3bcf430fc8dced32d4c05e0/python/arcticdb/version_store/library.py#L1537) we do:

```python
index = NameWithDType(info["col_names"]["index"], info["col_names"]["index_dtype"])
```

where `info["col_names"]["index_dtype"]` comes from [`_store.py`](https://github.com/man-group/ArcticDB/blob/904b0e5b88bdf9b7c3bcf430fc8dced32d4c05e0/python/arcticdb/version_store/_store.py#L2568) here:

```python
index_dtype = []
...
if input_type == "df":
    ...
    if index_type == "multi_index" or (index_type == "index" and index_metadata.is_not_range_index):
        ...
        index_dtype = [dtypes.pop(0)]
    else:
        # RangeIndex
        ...
        index_dtype = [dtype_desc]

    if index_type == "multi_index":
        for i in range(1, index_metadata.field_count + 1):
            ...
            index_dtype.append(dtypes.pop(0))
...
return {
    "col_names": {"columns": columns, "index": index, "index_dtype": index_dtype},
    ...
```

So therefore I'd suggest we keep the current API, but have the `dtype` optionally be a `List[TypeDescptor]` since to support the case of multi-level indexes, the `index` property of `SymbolDescriptor` is always a list.

```suggestion
    dtype: Union[TypeDescriptor, List[TypeDescriptor]]
```